### ISSUE TITLE
Bintray deprecation and cocoapods dependencies

### ISSUE THREAD

**Issue by polmum** [googleads/googleads-mobile-ios-mediation#311]
Hey!
As you most likely know, Bintray is being deprecated on May 1st. I was wondering if you were going to update the podspecs that are still using Bintray as the source of the SDK. The ones that I found are still doing this are GoogleMobileAdsMediationTapjoy, GoogleMobileAdsMediationChartboost and GoogleMobileAdsMediationFyber.

**Comment by ericleich** [comment]
It's on our radar and definitely a goal to have at least 1 release for every adapter off of bintray. We haven't experimented with old releases trying to delete/recreate the podspec, and may not until May 1st. CocoaPods has [documentation](https://guides.cocoapods.org/terminal/commands.html#pod_trunk_delete) saying that once you delete, you can never re-update that version. But since they'll no longer be functional, it'll be worth a shot to try. If not, we will have to mark deprecated.

**Comment by polmum** [comment]
Yes, I meant fixing it with new releases using your new source for the SDKs, so good to know!
About recreating the same podspec versions, in my experience it works partially. The main issue is that when a client that had the older version does a `pod update`, cocoapods does not replace the podspec with the newer version, most likely because the spec is cached locally. So clients are forced to delete the local cache or repository for it to work.

**Comment by ericleich** [comment]
Just tested a delete/re-add on an old Facebook adapter version and CocoaPods did appear to let me reupload:

https://github.com/CocoaPods/Specs/commit/504439327c91e29c5831ea2afba0f3b24c3d365e

https://github.com/CocoaPods/Specs/commit/6058503d12b9edb034468865f30b0f38e1ff582d

If a pod is locally cached, I suspect that'd be fine -- the new zip should be functionally equivalent to the old one.

So aside from making new releases of remaining adapters, I do plan to backport some recent versions of adapters as well.

**Comment by polmum** [comment]
If the pod is cached, you are right, there should be no problems. I think there is a another case though, when the pod is not cached but the old podspec is the one in the local repo that cocoapods creates. In this case, doing a pod install would use the old pod, which will fail for using bintray as its source. And a pod repo update would not redownload the new podspec with the updated URL.
I could be wrong though, but I think that this is what would happen in this case.

**Comment by ericleich** [comment]
@polmum you're right, that does sound plausible. I don't see anything we can do at this point to address that, though.

**Comment by polmum** [comment]
Nope, not really much you can do. That's why I think if you update older podspecs, it should only be done as a mitigation to the problem, not an actual solution. I would also recommend to always push a new podspec's version with the new source of the latest SDK, even if the version does not really have new code.

**Comment by ericleich** [comment]
Could replace version x.y.z.a with a new x.y.z.a.0 new version for each outdated version, but [pod trunk deprecate](https://guides.cocoapods.org/terminal/commands.html#pod_trunk_deprecate) looks to only support the full pod though, not per version. So my only option would be to leave x.y.z.a broken, or delete it. In this special case, I think updating the spec in place will result in less breakages than those other 2 options.
